Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/PolDiffILLReduction.py

```diff
@@ -0,0 +1,876 @@
+                             direction=Direction.Input,
+                             doc="Whether to perform component separation and what type of method to use.")
+
+        self.declareProperty(name="SampleGeometry",
+                             defaultValue="None",
+                             validator=StringListValidator(["None", "FlatPlate", "Cylinder", "Annulus", "Custom"]),
+                             direction=Direction.Input,
+                             doc="Sample geometry for self-attenuation correction to be applied.")
+
+        self.setPropertySettings('SampleGeometry', EnabledWhenProperty(vanadium, sample, LogicOperator.Or))
+
+        self.declareProperty(PropertyManagerProperty('ExperimentPropertiesDictionary', dict()),
+                             doc="Dictionary for the geometry used for self-attenuation correction.")
+
+        self.setPropertySettings('ExperimentPropertiesDictionary',
+                                 EnabledWhenProperty('SampleGeometry', PropertyCriterion.IsNotEqualTo, 'Custom'))
+
+        self.declareProperty(FileProperty(name="SampleGeometryFile",
+                             defaultValue="",
+                             action=FileAction.OptionalLoad),
+                             doc="The path to the custom geometry for self-attenuation correction to be applied.")
+
+        self.setPropertySettings('SampleGeometryFile', EnabledWhenProperty('SampleGeometry', PropertyCriterion.IsEqualTo, 'Custom'))
+
+        self.declareProperty(name="DetectorEfficiencyCalibration",
+                             defaultValue="None",
+                             validator=StringListValidator(["None", "Vanadium", "Incoherent",  "Paramagnetic"]),
+                             direction=Direction.Input,
+                             doc="Detector efficiency calibration type.")
+
+        self.setPropertySettings('DetectorEfficiencyCalibration', sample)
+
+        self.declareProperty(name="IncoherentCrossSection",
+                             defaultValue=0,
+                             validator=IntBoundedValidator(lower=0),
+                             direction=Direction.Input,
+                             doc="Cross-section for incoherent scattering, necessary for setting the output on the absolute scale.")
+
+        incoherent = EnabledWhenProperty('DetectorEfficiencyCalibration', PropertyCriterion.IsEqualTo, 'Incoherent')
+
+        absoluteNormalisation = EnabledWhenProperty('AbsoluteUnitsNormalisation', PropertyCriterion.IsDefault)
+
+        self.setPropertySettings('IncoherentCrossSection', EnabledWhenProperty(incoherent, absoluteNormalisation,
+                                                                               LogicOperator.And))
+
+        self.declareProperty(name="OutputUnits",
+                             defaultValue="Wavelength/TOF",
+                             validator=StringListValidator(["Wavelength/TOF", "TwoTheta", "Q"]),
```

Comment:
I'm confused about wavelength/TOF. Why is that in the list?

Reply:
That's the default input unit, either in units of wavelength or the provided TOF unit, that will be kept in the output.